src/main/java/net/neoforged/neoforge/attachment/AttachmentSyncReason.java

```diff
@@ -0,0 +1,7 @@
+package net.neoforged.neoforge.attachment;
+
+public enum AttachmentSyncReason {
```

Comment:
This, as it exists now, doesn't seem _particularly_ valuable. To end users, it only propagates into `IAttachmentSyncHandler#write` (and not `#read`), neither of which should change the data written based on the condition of the entity.

The only usage I could see for this is reacting to the first time the data was received on the client, but none of the fixtures even allow for that (doing so at deserialization time is inadvisable).